MapboxCoreNavigation/RouteController.swift

```diff
@@ -234,22 +234,28 @@ extension RouteController: CLLocationManagerDelegate {
         if let upComingStep = routeProgress.currentLegProgress.upComingStep {
             let isCloseToUpComingStep = newLocation.isWithin(radius, of: upComingStep)
             if !isCloseToCurrentStep && isCloseToUpComingStep {
-                
-                // Increment the step
-                routeProgress.currentLegProgress.stepIndex += 1
-                
-                // and reset the alert level since we're on the next step
                 let userSnapToStepDistanceFromManeuver = distance(along: routeProgress.currentLegProgress.currentStep.coordinates!, from: location.coordinate)
```

Comment:
I don’t think it’s correct to calculate this distance from the beginning of the current step, which could be many miles behind the user at this point. It could well be correct to increment `stepIndex` after this point instead of before it, but then this line should use `upcomingStep` instead of `currentStep`.